db/src/commonMain/kotlin/com/islamversity/db/model/Calligraphy.kt

```diff
@@ -53,13 +57,17 @@ data class Calligraphy(
          */
         operator fun invoke(code: String): Calligraphy {
             val codes = code.split("_")
-            if (codes.size < 2) {
-                throw IllegalArgumentException("a calligraphy is created using 3 parts: languageCode, levelType, calligraphyCode")
+            if (codes.isEmpty()) {
```

Comment:
In the [LanguageCode](url) the `isValid `condition has not been updated

```
     val isValid: Boolean
        get() =
            lang.length == 2 &&
                    lang == lang.toLowerCase()
```

Reply:
please consider nullable `CalligraphyName`